Lib/getpass.py

```diff
@@ -68,17 +70,41 @@ def unix_getpass(prompt='Password: ', stream=None):
                 old = termios.tcgetattr(fd)     # a copy to save
                 new = old[:]
                 new[3] &= ~termios.ECHO  # 3 == 'lflags'
+                if mask:
+                    new[3] &= ~termios.ICANON
```

Comment:
```suggestion
                    new[3] |= termios.ICANON
```

Shouldn't it be that way? also, shouldn't you use `ECHOE` so that previous characters are erased as well?

Reply:
This line is responsible for disabling buffering. when using your proposal mask chars will be displayed only after pressing `Enter`.